### PR TITLE
PLA 1536 composite gen readme

### PR DESCRIPTION
**Pull request by coolguru**
Add command for composite generate readme doc.

TODO
- Better error handling
- Write tests

### CODE REVIEW

File: cmd/composite_gendoc.go [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+	"strconv"
+
+	"github.com/CloudCoreo/cli/cmd/util"
+	"github.com/CloudCoreo/cli/cmd/content"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+var docHeaders =  map[string]string {
+		"head.md": "",
```
Comment (pauldallen) [inline cmd/composite_gendoc.go]: does this need a config.rb too then?

File: cmd/composite_gendoc.go [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+		// check if
+		if check(option.required, option.defaultValues) {
+			counter ++
+			contentBytes.WriteString("### `"+ option.key + "`:\n")
+			contentBytes.WriteString("  * description: "+ option.description)
+
+			if printVar && option.defaultValues != nil{
+
+				switch option.valueType  {
+				case "boolean":
```
Comment (pauldallen) [inline cmd/composite_gendoc.go]: we need the "case" type as well - 
check this out:
https://github.com/CloudCoreo/etcd-cluster/blob/master/config.yaml#L43

File: cmd/content/const.go [hunk 3]
```diff
@@ -28,6 +28,12 @@ const (
 
 	// errors
 	ERROR_MISSING_API_KEY_SECRET_KEY = "Missing API key or/and Secret key. Please run 'Coreo configure' to configure them."
+	ERROR_MISSING_FILE = "Missing %s file, creating it."
+
+	//Flags
+	CMD_FLAG_DIRECTORY_LONG = "directory"
+	CMD_FLAG_DIRECTORY_SHORT = "D"
```
Comment (pauldallen) [inline cmd/content/const.go]: lower case d

File: cmd/version.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	RootCmd.AddCommand(versionCmd)
+}
+
+var versionCmd = &cobra.Command{
+	Use:   "version",
+	Short: "Print the version number of Coreo CLI",
```
Comment (pauldallen) [inline cmd/version.go]: extract strings??

File: cmd/composite_gendoc.go (hunk 1 above)
Comment (coolguru) [inline cmd/composite_gendoc.go]: file name here are just for files that needs to be included in readme file

File: cmd/content/const.go (hunk 3 above)
Comment (coolguru) [inline cmd/content/const.go]: resolved

**Pull request merged**